README.md

```diff
@@ -112,84 +127,14 @@ The IDE server should manage the lifecycle of the GraphQL server. Ideally, the I
- * {
- *    protocol: 'graphql-protocol',
- *    id: number,
- *    method: string, // one of the function names below, e.g. `getDiagnostics`
- *    args: {
- *      query?: string,
- *      position?: Point,
- *      filePath?: Uri,
- *    }
- * }
- */
-// Diagnostics (lint/validation)
-export type GraphQLDiagnosticMessage = {
-  name: string,
-  type: string,
-  text: string,
-  range: atom$Range,
-  filePath: string,
-};
-
-export function getDiagnostics(
-  query: string,
-  filePath: Uri,
-) : Promise<Array<GraphQLDiagnosticMessage>> {
-  throw new Error('RPC stub');
-}
-
-// Autocomplete  Suggestions (typeahead)
-export type GraphQLAutocompleteSuggestionType = {
-  text: string,
-  typeName: ?string,
-  description: ?string,
-};
-
-export function getAutocompleteSuggestions(
-  query: string,
-  position: atom$Point,
-  filePath: Uri,
-) : Promise<Array<GraphQLAutocompleteSuggestionType>> {
-  throw new Error('RPC stub');
-}
+GraphQL Language Server uses [JSON-RPC](http://www.jsonrpc.org/specification) to communicate with the IDE servers to perform language service features. The language server currently supports two communication transports: Stream (stdio) and IPC. For IPC transport, the reference guide to be used for development is [the language server protocol](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md) documentation.
 
-// Definitions (hyperlink)
-export type Definition = {
-  path: Uri,
-  position: Point,
-  range?: Range,
-  id?: string,
-  name?: string,
-  language: string,
-  projectRoot?: Uri,
-};
-
-export type DefinitionQueryResult = {
-  queryRange: Array<Range>,
-  definitions: Array<Definition>,
-};
-
-export function getDefinition(
-  query: string,
-  position: atom$Point,
-  filePath: Uri,
-): Promise<DefinitionQueryResult> {
-  throw new Error('RPC stub');
-}
+For each transports, there is a slight difference between both JSON message format, especially in how the methods to be invoked are defined - below are the currently supported methods for each transports (will be updated as progresses are made):
 
-// Outline view
-export function getOutline(query: string): Promise<Outline> {
-  throw new Error('RPC stub');
-}
-
-// Disconnect signal - gracefully terminate the connection on IDE exit
-export function disconnect(): void {
-  throw new Error('RPC stub');
-}
-```
+|                     | Stream                       | IPC                               |
+| -------------------:|------------------------------|-----------------------------------|
+| Diagnostics         | `getDiagnostics`             | `textDocument/publishDiagnostics` |
+| Autocompletion      | `getAutocompleteSuggestions` | `textDocument/completion`         |
+| Outline             | `getOutline`                 | Not supported yet                 |
+| Go-to definition    | `getDefinition`              | Not supported yet                 |
```

Comment:
Yeah, I know it will render correctly to HTML, but one of the ideas of Markdown is that the plain-text should look good too. I personally think its ok if the lines here are long (in fact, we're already not hard-wrapping the lines in the file, so we already have lots of long lines).